File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/controller/StoreController.java
package com.pds.localpos.userservice.controller;

import com.pds.localpos.userservice.dto.response.StoreResponse;
import com.pds.localpos.userservice.service.StoreService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Set;

@Slf4j
@RestController
@RequestMapping("/api/v1/stores")
@RequiredArgsConstructor
public class StoreController {

    private final StoreService storeService;

    @GetMapping
    public ResponseEntity<Set<StoreResponse>> getAllStores() {
        log.info("GET /stores - Fetching all stores");
        Set<StoreResponse> stores = storeService.getAllStores();
        log.info("Fetched {} stores", stores.size());
        return ResponseEntity.ok(stores);
    }
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/controller/StoreController.kt
package com.pds.localpos.userservice.controller

import com.pds.localpos.userservice.dto.StoreDTO
import com.pds.localpos.userservice.service.StoreService
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/stores")
class StoreController(
    private val storeService: StoreService
) {

    private val logger = LoggerFactory.getLogger(StoreController::class.java)

    @GetMapping
    fun getAllStores(): ResponseEntity<Set<StoreDTO>> {
        logger.info("Fetching all stores")
        val stores = storeService.getAllStores()
        logger.info("Found {} stores", stores.size)
        return ResponseEntity.ok(stores)
    }
}
